packages/extension/src/ui/features/actions/ApproveTransactionScreen.tsx

```diff
@@ -281,7 +280,7 @@ export const ApproveTransactionScreen: FC<ApproveTransactionScreenProps> = ({
   }, [selectedAccount, buildResult, onSubmit, onReject, navigate])
 
   if (!selectedAccount) {
-    return <Navigate to={routes.accounts()} />
```

Comment:
This should not happen, since when approving tx it should always have an account.

Previously it would end up in a loop since when navigating to account path, it goes through the AppRoute, which would check actions. Because action is still there, it would go back to the approve transaction screen and end up here again.

Reply:
Why this branch was hit before? The selected account should always be defined, right?

Reply:
it is only hit when we are in one network, but interact with dApp in another network. my guess is that we are having more users and they connect to more more dApps in different networks, when they switch between them they would hit this.

Reply:
I see. Makes sense.